test-apps/tutorial-servo-can-control/src/TutorialServo.cpp

```diff
@@ -0,0 +1,37 @@
+#include "TutorialServo.h"
+#include "mbed.h"
+
+//I believe we are writing to servoPin, which I set to be PA_1
+CAN can1(PA_0, servoPin);
+float pulseWidthToSend;
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees = 180.0, float minPulsewidthInMs = 1, float maxPulsewidthInMs = 2) {
+    m_servoPwmOut = servoPin;
+    m_servoRangeInDegrees = servoRangeInDegrees;
+    m_minPulsewidthInMs = minPulsewidthInMs;
+    m_maxPulsewidthInMs = maxPulsewidthInMs;
+}
+
+// Set servo position (ex. 45 deg)
+void setPositionInDegrees( const float degrees) {
+    m_servoPwmOut.period_ms(20);
+    pulseWidthToSend = (1+degrees/servoRangeInDegrees) / 1000;
+    if(can1.write(CANMessage(1337. &pulseWidthToSend, 4))) {
+        printf("Sending through CAN to servo!");
+    }
```

Comment:
We're not actually commanding the servo via CAN, we're commanding it through a PWM signal (using the `m_servoPwmOut` pin). We control the servo position through the PWM pulse width. You'll want to map from degrees to [`m_minPulsewidthInMs`, `m_maxPulsewidthInMs`].

Reply:
Attempted to fix in 0fabe3f72fab676fa8c9a4c6ba2574a79b1cc32f